spring-auto-restdocs-json-doclet-jdk9 throw java.lang.ClassCastException

env versions

openjdk 11.0.2
spring-auto-restdocs-core  2.0.6
maven-javadoc-plugin  3.0.1
spring-restdocs-mockmvc  2.0.3.RELEASE

err log 
```
[ERROR] java.lang.ClassCastException: class com.sun.tools.javac.tree.DCTree$DCErroneous cannot be cast to class com.sun.source.doctree.BlockTagTree (com.sun.tools.javac.tree.DCTree$DCErroneous and com.sun.source.doctree.BlockTagTree are in module jdk.compiler of loader 'app')
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.lambda$fromMethodDoc$0(MethodDocumentation.java:57)
[ERROR] at java.base/java.lang.Iterable.forEach(Iterable.java:75)
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.lambda$fromMethodDoc$1(MethodDocumentation.java:51)
[ERROR] at java.base/java.util.Optional.ifPresent(Optional.java:183)
[ERROR] at capital.scalable.restdocs.jsondoclet.MethodDocumentation.fromMethodDoc(MethodDocumentation.java:51)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.addMethod(ClassDocumentation.java:68)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.lambda$fromClassDoc$0(ClassDocumentation.java:50)
[ERROR] at java.base/java.lang.Iterable.forEach(Iterable.java:75)
[ERROR] at capital.scalable.restdocs.jsondoclet.ClassDocumentation.fromClassDoc(ClassDocumentation.java:45)
[ERROR] at capital.scalable.restdocs.jsondoclet.ExtractDocumentationAsJsonDoclet.lambda$run$1(ExtractDocumentationAsJsonDoclet.java:68)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
[ERROR] at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
[ERROR] at java.base/java.util.Iterator.forEachRemaining(Iterator.java:133)
[ERROR] at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
[ERROR] at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
[ERROR] at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
[ERROR] at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
[ERROR] at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
[ERROR] at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
[ERROR] at capital.scalable.restdocs.jsondoclet.ExtractDocumentationAsJsonDoclet.run(ExtractDocumentationAsJsonDoclet.java:63)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.parseAndExecute(Start.java:582)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:431)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Start.begin(Start.java:344)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Main.execute(Main.java:63)
[ERROR] at jdk.javadoc/jdk.javadoc.internal.tool.Main.main(Main.java:52)
```

--- a/pom.xml
+++ b/pom.xml
@@ -339,6 +339,11 @@
                 <version>3.9</version>
             </dependency>
             <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-text</artifactId>
+                <version>1.8</version>
+            </dependency>
+            <dependency>
                 <groupId>commons-io</groupId>
                 <artifactId>commons-io</artifactId>
                 <version>2.6</version>

--- a/spring-auto-restdocs-json-doclet-jdk9/pom.xml
+++ b/spring-auto-restdocs-json-doclet-jdk9/pom.xml
@@ -26,7 +26,7 @@
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>
-            <artifactId>commons-lang3</artifactId>
+            <artifactId>commons-text</artifactId>
         </dependency>
         <dependency>
             <groupId>junit</groupId>

--- a/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/FieldDocumentation.java
+++ b/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/FieldDocumentation.java
@@ -2,7 +2,7 @@
  * #%L
  * Spring Auto REST Docs Json Doclet for JDK9+
  * %%
- * Copyright (C) 2015 - 2019 Scalable Capital GmbH
+ * Copyright (C) 2015 - 2020 Scalable Capital GmbH
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -29,6 +29,7 @@ import java.util.Map;
 import java.util.Optional;
 
 import com.sun.source.doctree.BlockTagTree;
+import com.sun.source.doctree.DocTree;
 import jdk.javadoc.doclet.DocletEnvironment;
 
 public class FieldDocumentation {
@@ -36,19 +37,23 @@ public class FieldDocumentation {
     private String comment;
     private final Map<String, String> tags = new HashMap<>();
 
+    private void addTag(DocTree tag) {
+        if (tag instanceof BlockTagTree) {
+            tags.put(
+                    cleanupTagName(((BlockTagTree) tag).getTagName()),
+                    cleanupTagValue(tag.toString()));
+        }
+    }
+
     public static FieldDocumentation fromFieldDoc(DocletEnvironment docEnv,
             Element fieldElement) {
         FieldDocumentation fd = new FieldDocumentation();
         fd.comment = cleanupDocComment(docEnv.getElementUtils().getDocComment(fieldElement));
 
-
         Optional.ofNullable(docEnv.getDocTrees().getDocCommentTree(fieldElement))
                 .ifPresent(docCommentTree -> docCommentTree.getBlockTags()
-                        .forEach(tag -> fd.tags.put(
-                                cleanupTagName(((BlockTagTree) tag).getTagName()),
-                                cleanupTagValue(tag.toString()))));
+                        .forEach(tag -> fd.addTag(tag)));
 
         return fd;
     }
-
 }

--- a/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/MethodDocumentation.java
+++ b/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/MethodDocumentation.java
@@ -50,7 +50,6 @@ public class MethodDocumentation {
 
         Optional.ofNullable(docEnv.getDocTrees().getDocCommentTree(methodElement))
                 .ifPresent(docCommentTree -> docCommentTree.getBlockTags().forEach(tag -> {
-                    System.out.println(tag.getKind());
                     if (tag.getKind().equals(DocTree.Kind.PARAM)) {
                         ParamTree paramTag = (ParamTree) tag;
                         md.parameters.put(paramTag.getName().toString(),

--- a/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/MethodDocumentation.java
+++ b/spring-auto-restdocs-json-doclet-jdk9/src/main/java/capital/scalable/restdocs/jsondoclet/MethodDocumentation.java
@@ -2,7 +2,7 @@
  * #%L
  * Spring Auto REST Docs Json Doclet for JDK9+
  * %%
- * Copyright (C) 2015 - 2019 Scalable Capital GmbH
+ * Copyright (C) 2015 - 2020 Scalable Capital GmbH
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -22,7 +22,7 @@ package capital.scalable.restdocs.jsondoclet;
 import static capital.scalable.restdocs.jsondoclet.DocletUtils.cleanupDocComment;
 import static capital.scalable.restdocs.jsondoclet.DocletUtils.cleanupTagName;
 import static capital.scalable.restdocs.jsondoclet.DocletUtils.cleanupTagValue;
-import static org.apache.commons.lang3.StringEscapeUtils.unescapeJava;
+import static org.apache.commons.text.StringEscapeUtils.unescapeJava;
 
 import javax.lang.model.element.Element;
 import java.util.HashMap;
@@ -50,11 +50,12 @@ public class MethodDocumentation {
 
         Optional.ofNullable(docEnv.getDocTrees().getDocCommentTree(methodElement))
                 .ifPresent(docCommentTree -> docCommentTree.getBlockTags().forEach(tag -> {
+                    System.out.println(tag.getKind());
                     if (tag.getKind().equals(DocTree.Kind.PARAM)) {
                         ParamTree paramTag = (ParamTree) tag;
                         md.parameters.put(paramTag.getName().toString(),
                                 unescapeJava(paramTag.getDescription().toString()));
-                    } else {
+                    } else if (tag instanceof BlockTagTree) {
                         md.tags.put(
                                 cleanupTagName(((BlockTagTree) tag).getTagName()),
                                 unescapeJava(cleanupTagValue(tag.toString())));

--- a/spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/DocumentedClass.java
+++ b/spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/DocumentedClass.java
@@ -54,6 +54,7 @@ class DocumentedClass {
      * @param force true if force (テスト)
      * @title Do initiate (テスト)
      * @deprecated use other method (テスト)
+     * @ Don't fail on invalid tags
      */
     private void initiate(String when, boolean force, int notDocumented) {
     }

--- a/spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/DocumentedClass.java
+++ b/spring-auto-restdocs-json-doclet-jdk9/src/test/java/capital/scalable/restdocs/jsondoclet/DocumentedClass.java
@@ -2,7 +2,7 @@
  * #%L
  * Spring Auto REST Docs Json Doclet for JDK9+
  * %%
- * Copyright (C) 2015 - 2019 Scalable Capital GmbH
+ * Copyright (C) 2015 - 2020 Scalable Capital GmbH
  * %%
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -23,6 +23,8 @@ import java.math.BigDecimal;
 
 /**
  * This is a test class. UTF 8 test: 我能吞下玻璃而不伤身体。Árvíztűrő tükörfúrógép
+ *
+ * @ Don't fail on invalid tags
  */
 class DocumentedClass {
     /**
@@ -34,6 +36,8 @@ class DocumentedClass {
 
     /**
      * Path within location
+     *
+     * @ Don't fail on invalid tags
      */
     private BigDecimal path;
 

